### DISCUSSION TITLE
Pico W for extras?

### DISCUSSION THREAD

**Discussion by mwedmark** [frntc/SIDKick-pico#6] [Ideas]
Does the implementation allow for some extras with WIFI and/or Bluetooth functionality in the future?
Any reason for a PiPico W not to work?

For example, some ideas:
- Sound output through BT directly to speaker?
- Web-UI for changing settings live while playing?
- Output to external WIFI board (f.e. ESP8266) to control LED's gor graphics feedback?
- Mouse control via either websocket or via BT-mouse?

**Comment by magwed** [comment]
Also this would make it possible to send the input/control values to SID to external listener for debugging/playing on f.e. a PC for reference! :-)

**Comment by frntc** [comment]
I have thought about streaming audio via BT, WLAN or USB (as a USB-microphone), but not about the other options. I might look into that at some point, but I don't know yet whether any/which of these functionalities impact the running bus communication and how much resources they consume (and when/if I might be trying things out, streaming audio would be higher on the list than others).

About the mouse: supporting USB mice is something I have in mind since years (already for the large SIDKick), but the "annoying" part is that mouse buttons require connections to the joystick port/CIA, i.e. it requires additional connections inside the C64 (and thus GPIOs).

**Reply by magwed** [reply]
Great. Are you using both cores already? Otherwise one core could be used for these extra tasks I guess. Depending if you are using some shared resource like DMA/PIO's already which would collide or be congested.

Aha I didn't think about that.. the buttons are, of course, handled completly separatly. I also build/use versions of the Unijoysticle which is exactly that => An external device hooked to the josytick ports. With the correct version you can imitate a C64 mouse using a BT-mouse, because from that side it can both emulate the analog part and the digital joystick/mouse buttons.
Anywawy, thanks for sharing! I'm building a couple of SKP's and are going to try with PicoW which I happen to have at hand. Just need some different 2.54mm pin versions first.

**Comment by mwedmark** [comment]
Maybe even MIDI through either WIFI and/or Bluetooth MIDI profile would be possible. Of course only MIDI which MIDI-IN and it would probably not be compatible with Cartridge-based ones.. still a cool thought.

I just soldered a Pico-W to my first SKP. I guess I need to setup the tool-chain and produce my own compatible firmware. Hopefully that is straight-forward. More so when not using either BT or WIFI. Maybe FW files could be produced for these as well in the future?

**Reply by frntc** [reply]
I was thinking about MIDI, but rather separated from the SID-emulation (these RP2040 are really cheap...). I'd prefer keeping the main loops of the emulation and bus communication simple this time. Those for the large, Teensy-based SIDKick are really hard to maintain.
